### PR TITLE
EID-1100: Handle non standard HTTP methods.

### PR DESCRIPTION
**Pull request by tomrosier**

### CODE REVIEW

File: lib/store_session_id.rb [hunk 1]
```diff
@@ -4,10 +4,15 @@ def initialize(app)
   end
 
   def call(env)
+    if ["POST", "GET"].exclude? env['REQUEST_METHOD']
```
Comment (jakubmiarka) [inline lib/store_session_id.rb]: linter is going to shout because of the array syntax. Plus we also use PUT method for our ajax calls...

Comment (tomrosier) [reply lib/store_session_id.rb]: Good shout 👍

Comment (jakubmiarka) [reply lib/store_session_id.rb]: there might a better way of doing it. It doesn't feel right listing the 'allowed' methods.. Why we don't just capture the unsupported ones?

Comment (jakubmiarka) [reply lib/store_session_id.rb]: In that case I would be tempted to do the whitelist on the server (nginx) and not even let the request to reach the app? 🤷‍♂️

Comment (tomrosier) [reply lib/store_session_id.rb]: @jakubmiarka would this be preferred? https://github.com/alphagov/verify-puppet/pull/1460/files

**Comment by tomrosier** [comment]
Abandoned, in-favour of nginx change.

**Pull request closed without merging**